src/docdb/tree/DocDBStoredProceduresTreeItem.ts

```diff
@@ -69,14 +77,22 @@ export class DocDBStoredProceduresTreeItem extends DocDBTreeItemBase<StoredProce
         return this.parent.getContainerClient(client);
     }
 
-    private validateName(name: string): string | null | undefined {
+    private async validateStoredProcedureName(name: string, currStoredProcedureNames: String[]): Promise<string | null | undefined> {
         if (name) {
-            if (name.indexOf("/") !== -1 || name.indexOf("\\") !== -1 || name.indexOf("?") !== -1 || name.indexOf("#") !== -1) {
-                return "Id contains illegal chars: /,\\,?,#";
+            if (name.length < 1 || name.length > 255) {
+                return "Name has to be between 1 and 255 chars long";
+            }
+            if (/[/\\?#&]/.test(name)) {
+                return localize("illegalChars", "Name contains illegal chars: /,\\,?,#,&");
             }
             if (name[name.length - 1] === " ") {
-                return "Id ends with a space.";
+                return localize("endsWithSpace", "Name cannot end with a space.");
+            }
+            if (currStoredProcedureNames.includes(name)) {
+                return localize('NameExists', 'Stored Procedure "{0}" already exists.', name);
             }
+        } else {
+            return localize("noEmptyValue", "Name cannot be empty");
```

Comment:
We can get rid of this because it's already covered by the `if (name.length < 1 || name.length > 255)` check

Reply:
I think we can remove `name.length < 1` from the first condition as "noEmptyValue" covers it. But checking for length will require `name` to not be null.